pkg/managers/watchermanager/watchermanager.go

```diff
@@ -73,8 +73,8 @@ func (wm *WatcherManager) runWatcher(ctx context.Context, w IWatcher) error {
 		case <-ticker.C:
 			err := w.Refresh(ctx)
 			if err != nil {
-				wm.l.Error("refresh failed", zap.Error(err))
-				return err
+				wm.l.Warn("refresh failed", zap.Error(err))
```

Comment:
This needs to be more nuanced than a catch-all. Not all errors should be treated same, for example:
- if `veth` watcher has unrecoverable failure, that's a Fatal error IMO
- if apiserver watcher has failure connecting to ApiServer, it could be intermittent

I would expect the watchers themselves to determine what is fatal and what is not, and determine whether to suppress the error or send it back up to this manager. The main watcher controller should return error if child watchers return error.